mslib/msui/mpl_pathinteractor.py

```diff
@@ -407,8 +397,467 @@ def __init__(self, ax=None, waypoints=None, mplpath=None,
+        for i, wpd in enumerate(waypoints_model_data):
+            textlabel = str(i)
+            if wpd.location != "":
+                textlabel = f"{wpd.location}"
+            text = self.ax.text(
+                x[i] + label_offset, y[i] + label_offset, textlabel,
+                bbox={"boxstyle": "round", "facecolor": "white", "alpha": 0.6, "edgecolor": "none"},
+                fontweight="bold", zorder=4, animated=True, clip_on=True,
+                visible=self.showverts and self.label_waypoints)
+            self.wp_labels.append(text)
+
+        # Redraw the artists.
+        if self.background:
+            self.canvas.restore_region(self.background)
+        try:
+            self.ax.draw_artist(self.pathpatch)
+        except ValueError as error:
+            logging.debug("ValueError Exception '%s'", error)
+        self.ax.draw_artist(self.wp_scatter)
+
+        for t in self.wp_labels:
+            self.ax.draw_artist(t)
+        if self.show_tangent_points:
+            self.ax.draw_artist(self.tangent_lines)
+        if self.show_solar_angle is not None:
+            self.ax.draw_artist(self.solar_lines)
+        self.canvas.blit(self.ax.bbox)
+
+    def draw_callback(self, event):
+        """Extends PathInteractor.draw_callback() by drawing the scatter
+           instance.
+        """
+        super().draw_callback(event)
+        if self.wp_scatter:
+            self.ax.draw_artist(self.wp_scatter)
+        if self.show_solar_angle:
+            self.ax.draw_artist(self.solar_lines)
+        if self.show_tangent_points:
+            self.ax.draw_artist(self.tangent_lines)
+
+    def set_path_color(self, line_color=None, marker_facecolor=None,
+                       patch_facecolor=None):
+        """Set the color of the path patch elements.
+        Arguments (options):
+        line_color -- color of the path line
+        marker_facecolor -- color of the waypoints
+        patch_facecolor -- color of the patch covering the path area
+        """
+        super().set_path_color(line_color, marker_facecolor,
+                               patch_facecolor)
+        if marker_facecolor is not None and self.wp_scatter is not None:
+            self.wp_scatter.set_facecolor(marker_facecolor)
+            self.wp_scatter.set_edgecolor(marker_facecolor)
+            self.markerfacecolor = marker_facecolor
+
+    def set_vertices_visible(self, showverts=True):
+        """Set the visibility of path vertices (the line plot).
+        """
+        super().set_vertices_visible(showverts)
+        if self.wp_scatter is not None:
+            self.wp_scatter.set_visible(self.show_marker)
+
+    def set_tangent_visible(self, visible):
+        self.show_tangent_points = visible
+
+    def set_solar_angle_visible(self, visible):
+        self.show_solar_angle = visible
+
+    def set_remote_sensing(self, ref):
+        self.remote_sensing = ref
+
+
+class PathV_GCPlotter(PathPlotter):
```

Comment:
The GC stands for "great circle". There are two, separate, path classes (probably for historical reasons; doesn't seem to fill any particular purpose). PathH and PathH_GC, which, presumably deal with horizontal paths in an Euclidian geometry, on the one hand, and on a sphere, on the other.

Please use the GC only for the Horizontal classes.

Reply:
Ah, got it.